template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateValidator.java

```diff
@@ -65,6 +68,21 @@ public void validateHostGroupScalingRequest(String accountId, Blueprint blueprin
         }
     }
 
+    private void validateNodeManagerAdjustment(HostGroup hostGroup, Integer adjustment, Collection<InstanceGroup> instanceGroups,
```

Comment:
We have this field `UpdateClusterV4Request.hostGroupAdjustment.forced`. Does the code checks if it is a force request then proceed without this validation?

Reply:
It depends, in case of thrash icon the code goes this path, so the validation will be skipped if force is enabled: https://github.com/hortonworks/cloudbreak/blob/7dd704fbee7e72d7141e33e021c4b4f34ce85f8f/core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackOperationService.java#L116-L120

In case of normal downscale, we don't care about force flag, we will validate:
https://github.com/hortonworks/cloudbreak/blob/3e4b5600c66d548bd472cf9770a9ef90bf58a824/core/src/main/java/com/sequenceiq/cloudbreak/service/ClusterCommonService.java#L137-L160

As we discussed with @biharitomi it looks fine, because every other validation will run in a normal downscale scenario (regardless force flag is enabled or not).